include/opentracing/dynamic_load.h

```diff
@@ -37,10 +38,10 @@
 //   }
 extern "C" {
 #ifdef OPENTRACING_BUILD_DYNAMIC_LOADING
-int __attribute((weak))
-OpenTracingMakeTracerFactory(const char* opentracing_version,
-                             const void** error_category,
-                             void** tracer_factory);
+OPEN_TRACING_EXPORT int OPEN_TRACING_ATT
+    OpenTracingMakeTracerFactory(
+    const char* opentracing_version, const void** error_category,
+    void** tracer_factory);
```

Comment:
`OPEN_TRACING_ATT` doesn't expand to anything on windows, right? 

Suppose I have two tracing libraries libTracerA, libTracerB, and then I link them both into my app with the intention of switching between them based on a config -- would that give a multiple definition error on windows because both would define `OpenTracingMakeTracerFactory `?

The weak attribute was there to support that scenario. Could we maybe use [`__declspec(selectany)`](https://msdn.microsoft.com/en-us/library/5tkz6s71.aspx) for this?

Reply:
Indeed, OPEN_TRACING_ATT does not expand to anything on Windows.

if we take your example, why would I link directly with the two tracers? I'd rather use the dynamic loading mechanism to load the implementation, no?

Reply:
ok I understand now your point.

Unfortunately, weak symbols are not exposed by MSVC compiler. selectany works only on data items and not on methods.
https://docs.microsoft.com/en-us/cpp/cpp/selectany

However, we can use the */FORCE:MULTIPLE* link flag to force the link of the binary even if multiple definitions of a symbol exists
https://docs.microsoft.com/en-us/cpp/build/reference/force-force-file-output

Also, on windows, using dynamic linking, having the same symbols is not a problem.
This last point is important. If we take again the example, linking TracerA and TracerB statically with OT will cause duplication of the symbols of OT, among them the Global Tracer, which is not what we really want.

What do you think?

Reply:
Could you use selectany with a data variable that's a function pointer?


> If we take again the example, linking TracerA and TracerB statically with OT will cause duplication of the symbols of OT, among them the Global Tracer, which is not what we really want.

Static libraries are usually built as thin archives (i.e. without the dependent libraries linked in), so there wouldn't be duplication of the OT symbols since OT would be linked in by the app along with the tracing libraries.

Reply:
done.